ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/resources/KsqlResourceTest.java

```diff
@@ -1883,11 +1908,25 @@ private static void validateQueryDescription(
     assertThat(queryDescription.getOverriddenProperties(), is(overriddenProperties));
   }
 
+  private KsqlResource givenKsqlResource(final ServiceContext serviceContext) {
```

Comment:
I feel like, to get proper testing here we should pass in a `Supplier<ServiceContext>` so that we can ensure that if we issue to requests, we get different instances of service contexts each time.

Also we should have `setUpKsqlResource` call this method so that we don't have two palces were we need to update things like the `InjectorChain` if we update them.

Reply:
done